### DISCUSSION TITLE
Allow to run ssh commands to localhost

### DISCUSSION THREAD

**Discussion by pierrelemee** [spatie/ssh#109] [Ideas]
# Allow to run ssh commands to localhost

TL;DR: as a user I might want to run SSH commands to a local Docker container. For this, I would propose to add a configuration method `allowLocalConnection`

I'm trying to execute a set of [SSH commands onto a server through Castor 🦫 ](https://castor.jolicode.com/going-further/helpers/ssh/). As it's aimed to configure a distant server, I'm testing it on a local Docker container (basically a debian base image with the relevant ssh config on top of it) to be able to wipe and start over (using the `--force-recreate` Docker option).

However, when testing I faced a "Permission denied" error that, after digging, revealed to be caused by the [local command override in the `Ssh::getExecuteCommand()` method](https://github.com/spatie/ssh/blob/main/src/Ssh.php#L172).

Although I understand this check, as a developer I would like to be allowed to overcome this by allowing local connection using a `Ssh::allowLocalConnection` method, like it's done for host check or password authentication. By default, the underlying variable would remain `false` to keep nominal behavior.

I can propose a patch for it :)
